src/controller/CHIPDeviceController.cpp

```diff
@@ -395,6 +429,21 @@ void DeviceController::ReleaseAllDevices()
     }
 }
 
+uint16_t DeviceController::FindDeviceIndex(SecureSessionHandle session)
```

Comment:
I know this is what the code used to look like with node id, but is there a reason this is not `FindDevice` returning `Device *`, with `nullptr` returned if not found?  Then we could use a ranged for loop instead of the indexing we are doing now, and callers wouldn't need to index back into the array; they would just use the pointer they got, if not null.

Reply:
While audit the usage of `FindDeviceIndex`, I don't think there is any reason preventing it from returning a `Device *`.

We can refactor it in another PR. but here I want to keep it consistent with `uint16_t DeviceController::FindDeviceIndex(NodeId id)`.

Reply:
Yep, followup is fine.  Just file an issue, please.